### PR TITLE
Enabled RBAC role assignment for disk encrpytion set key

### PR DESCRIPTION
**Pull request by AlexanderSehr**
# Description

- Enabled RBAC authorization of the Key used in the Disk Encryption Set. Before it was limited to RBAC.
> NOTE: The solution works with User-Assigned Identities only. The reason is that the System-Assigned identity is not yet available when the role assignment would need to happen

| Pipeline |
| - |
| [Compute: DiskEncryptionSets](https://github.com/Azure/ResourceModules/actions/workflows/ms.compute.diskencryptionsets.yml) |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

### CODE REVIEW

File: modules/Microsoft.Compute/diskEncryptionSets/.bicep/nested_keyVaultPermissions.bicep [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    roleDefinitionId: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '12338af0-0e69-4776-bea7-57ae8d297424') // Key Vault Crypto User
+    principalType: 'ServicePrincipal'
+  }
+}
+
+// ============= //
+// Access Policy //
+// ============= //
+
+module keyVaultAccessPolicies '../../../Microsoft.KeyVault/vaults/accessPolicies/deploy.bicep' = {
```
Comment (ahmadabdalla) [inline modules/Microsoft.Compute/diskEncryptionSets/.bicep/nested_keyVaultPermissions.bicep:50]: @AlexanderSehr should this only run when `rbacAuthorizationEnabled` is `false`? As the default is `true` but this means that the access policy will run regardless. Am I understanding this correctly?

File: modules/Microsoft.Compute/diskEncryptionSets/.bicep/nested_roleAssignments.bicep [hunk 2]
```diff
@@ -69,7 +69,7 @@ var builtInRoleNames = {
 }
 
 resource diskEncryptionSet 'Microsoft.Compute/diskEncryptionSets@2020-12-01' existing = {
-  name: last(split(resourceId, '/'))
+  name: last(split(resourceId, '/'))!
```
Comment (ahmadabdalla) [inline modules/Microsoft.Compute/diskEncryptionSets/.bicep/nested_roleAssignments.bicep:72]: What is this used for?

File: modules/Microsoft.Compute/diskEncryptionSets/deploy.bicep [hunk 3]
```diff
@@ -60,11 +60,27 @@ resource defaultTelemetry 'Microsoft.Resources/deployments@2021-04-01' = if (ena
...
+resource keyVault 'Microsoft.KeyVault/vaults@2021-10-01' existing = {
+  name: last(split(keyVaultResourceId, '/'))!
   scope: resourceGroup(split(keyVaultResourceId, '/')[2], split(keyVaultResourceId, '/')[4])
+
+  resource key 'keys@2021-10-01' existing = {
+    name: keyName
+  }
 }
 
+// Note: This is only enabled for user-assigned identities and the service's system-assigned identity isn't available during its intitial deployment
```
Comment (ahmadabdalla) [inline modules/Microsoft.Compute/diskEncryptionSets/deploy.bicep:72]:
```suggestion
// Note: This is only enabled for user-assigned identities as the service's system-assigned identity isn't available during its initial deployment
```

File: modules/Microsoft.Compute/diskEncryptionSets/.bicep/nested_roleAssignments.bicep (hunk 2 above)
Comment (AlexanderSehr) [reply modules/Microsoft.Compute/diskEncryptionSets/.bicep/nested_roleAssignments.bicep:72]: Hey @ahmadabdalla this is the new character introduced by the PG to address this warning #2641 . The new Bicep version which is required for this change is rolled out to the GitHub agents now and hence I addressed the warning for my implementation here directly.

File: modules/Microsoft.Compute/diskEncryptionSets/.bicep/nested_keyVaultPermissions.bicep (hunk 1 above)
Comment (AlexanderSehr) [reply modules/Microsoft.Compute/diskEncryptionSets/.bicep/nested_keyVaultPermissions.bicep]: Good find. I accidently deleted the condition when replacing the plain `resource` deployment with the module. Should be fixed now.
Fun fact: This is how it was implemented originally. We borderline always set the access policies no matter if they are eventually used or not.

File: modules/Microsoft.Compute/diskEncryptionSets/.bicep/nested_roleAssignments.bicep (hunk 2 above)
Comment (ahmadabdalla) [reply modules/Microsoft.Compute/diskEncryptionSets/.bicep/nested_roleAssignments.bicep:72]: Thanks for clarifying @AlexanderSehr . Learned something new today 😀

**Review by ahmadabdalla — APPROVED** [review]

File: modules/Microsoft.Compute/diskEncryptionSets/.bicep/nested_roleAssignments.bicep (hunk 2 above)
Comment (AlexanderSehr) [reply modules/Microsoft.Compute/diskEncryptionSets/.bicep/nested_roleAssignments.bicep:72]: No problem. It's a nasty nasty world out there :D

**Pull request merged**